rust/src/tx_builder.rs

```diff
@@ -417,6 +428,21 @@ impl TransactionBuilder {
         self_copy.set_fee(&to_bignum(0x1_00_00_00_00));
         min_fee(&self_copy)
     }
+
+    fn minimum_utxo_val(&self) -> Result<Coin, JsError> {
+        if self
+            .outputs
+            .0
+            .iter()
+            .any(|output| output.amount().multiasset().is_some())
+        {
+            Err(JsError::from_str(
+                "Transaction with non-ada outputs is not implemented",
+            ))
+        } else {
+            Ok(self.constant_min_utxo_value)
```

Comment:
I would change this to just always return the mainnet min utxo value and add a comment that this will be replaced by formula once that formula is finalized.

Reply:
I'm not sure I understand, do you want to

a) remove the parameter in the txbuilder constructor and always return 1 (a few tests need to be fixed in that case,as they rely on it)
or
b) don't throw an error and just return 1 or 2. And also leave the comment about the formula.

Reply:
I wonder though, why can't we change the txbuilder to take the 3 constants of the formula in the spec as arguments and just compute the thing from there? Even if we don't have them right now, they are network parameters anyway.

Reply:
The thing is there are actually four parameters:

1) 3 parameters to calculate the approximate size of the transaction (outputSizeConstants in the spec)
2) One parameter to decide how many lovelaces each unit of this virtual size is worth (adaPerUTxOByte in the spec)

We could make the transaction builder take all four of these as parameters. I was thinking we could maybe just hardcode the three parameters of (1) and only take (2) as an argument, but if we think these will change based on the network maybe you're right we maybe should just create a struct to represent all of these and pass them into the tx builder. It would maybe also probably make it easier to port all the tests

I haven't been told what the values of the three parameters in (1) will be, only that the result of applying them will give 1 ADA when only ADA is used in the tx.

Reply:
I see, I missed the part about (2). That's what `ada_per_unit_size` means, then. And right now (1) is hardcoded to (1, 0, 0), essentially, although it just throws an error, so it is more like (1, `undefined`, `undefined`).
 
I don't mind hardcoding (1) too with the mainnet min utxo value if you want (although I don't know it), I just think it is more consistent to leave as it is, because it was already parametric.

Tbh I think just taking (1) for now it's fine but I don't know that well the how's this library used and how hard it is to make a breaking change later.

Reply:
Okay, what we have right now is a total hack, but adding these 4 parameters as an input to the TxBuilder requires refactoring things so if you want to just leave it as-is for now I'm okay with it (change `ada_per_unit_size` to a struct in a follow-up PR)